lib/http/HttpServer.ts

```diff
@@ -118,6 +118,7 @@ class HttpServer {
    */
   public close = () => {
     return new Promise<void>((resolve) => {
+      this.server.removeAllListeners();
```

Comment:
I figure if anything it makes more sense for this to be in the `close` callback below.